### ISSUE TITLE
Support using multiple repositories from the UI

### ISSUE THREAD

**Issue by maxime1992** [kopia/kopia#2976]
At the moment, it looks like we can only have one repository from the UI.

It'd be fantastic if **a snapshot had the repository defined in it instead of having one global repository**.

This would let us have any amount of snapshots, targeting any repository.

If personally find this the biggest issue to migrate fully to using Kopia instead of using Duplicati. I want to back things up on different repositories and having only 1 is quite limiting.

**Comment by basldfalksjdf** [comment]
You can use multiple repositories in KopiaUI. Right click the icon and select Connect To Another Repository.

**Comment by lupusA** [comment]
Support for multiple repositories within the web-ui is meant here, i guess. The clients support it, but not the native web-ui which you get when running kopia, for example, on docker.

Cheers,

**Comment by maxime1992** [comment]
Yup, to clarify I'm using docker here as I'm using it on a server (not only but that's one of my important use cases) :+1:

**Comment by Uberi-puzo** [comment]
Much needed. It should be a single command center in the main interface, not in the tray. This is done in all major backup programs, and it is convenient. Select a file, then select from several backup stores where this file is saved, and you can immediately see the version. Or, select a backup storage from the interface, and search for a file already in it.

This is just a request for improvement, but it would be great if it were implemented.

**Comment by ykktrcb7** [comment]
this is an important one

**Comment by Ra72xx** [comment]
I want to use Kopia for both local and cloud backups on a headless server. So it would be great to be able to configure multiple repositories from the WebUI. Running two Kopia instances in parallel to achieve this seems somehow clunky.

**Comment by ados8** [comment]
Absolutely need a central interface in the Kopia web-ui for managing multiple repositories.
Having to run multiple web-ui instances, one for each repo, is the opposite direction to how many other backup tools work.

**Comment by brainiac19** [comment]
Same. I understand it's already possible if set up correctly, but they are more like workarounds than solutions to me.

**Comment by ifooth** [comment]
feature request +1， when use velero backup, a k8s namespace is a velero repo, and i also have a local kopia repo

**Comment by abrahampm** [comment]
I would like to migrate my backups from Duplicati to Kopia but this is a huge shortcoming when running on Docker. It would be great to have this UI feature implemented.

**Comment by televisi** [comment]
What a great open source initiative!
Agree with this, need to be able to connect to multiple repositories through kopia GUI (installed on docker).

**Comment by MrEAlderson** [comment]
Kinda unexpected for kopia to favor a 1-1-1 over 3-2-1 backup rule

**Comment by statelycurmudgeon** [comment]
It’s really none of my business how the devs do their work, because I’m an incredibly happy user of the product and it functions marvelously. 

But… I use Kopia in docker, and I really want to be able to back up entire large zfs datasets to my remote NAS, and a subset to my rsync.net instance.  It seems like it’d be computationally less expensive to do the scan once but it also might be risky. And if I’m honest, I want to do cloud less frequently than remote NAS. But running two instances of Kopia feels “silly”. Maybe someone can explain to me why it’s logical - things often are.  But man I would love to get rid of one of my instances.

**Comment by danielschwab** [comment]
feature request +1

**Comment by derpderpington** [comment]
Much needed feature

**Comment by ggmoritz** [comment]
Most needed feature definitely

**Comment by janreinhardt** [comment]
want this feature, too.
3-2-1 backup rule needs 2 connected repositories and Kopia should be able to handle it. If Kopia could deal with multiple repos I would prefer it over Zerobyte (= Restic) and Duplicati and Borg.

**Comment by jumbi77** [comment]
+1, highly appreciate if web-GUI can handle multiple repositories.

**Comment by bjoernch** [comment]
Yes, please implement this feature — it’s essential. I switched from Duplicati due to backup issues; everything else works well, but this feature is a must-have for Kopia.

**Comment by essare** [comment]
+1 this is a much-needed feature

**Comment by pauvt** [comment]
+1 Super needed!

See how well it could be done: 

https://github.com/garethgeorge/backrest

**Comment by martxns** [comment]
Having recently switched from Duplicati to Kopia this shook me that it's not the norm to be able to see all repositories / backup locations from the UI.

Instead we need to either run multiple docker instances of Kopia to access each repository or to disconnect and reconnect from each repository to view backups from one machine (e.g. LAN and Cloud) please review this

**Comment by janreinhardt** [comment]
I have switched to Zerobyte because of this.

**Comment by Simplenuity** [comment]
> Having recently switched from Duplicati to Kopia 

May I ask why you switched?

(Am aware of that this should be asked outside of this thread, but am not aware of how.)

**Comment by martxns** [comment]
> > Having recently switched from Duplicati to Kopia
> 
> May I ask why you switched?
> 
> (Am aware of that this should be asked outside of this thread, but am not aware of how.)

Switched solely for the single docker container that can backup to local and offsite + cloud (3-2-1), as well as being able to be a backup target for another server that is running a docker Kopia container. 

It just made sense until I realised you can't view all repositories in the UI... I mean it's not broken, just very annoying to have to disconnect and reconnect to each repository to view and configure it via the GUI? Just let us see them all or switch between them easily without having to remember the logins.

Don't think there's an alternative that does all the above when compared to Kopia (that can make backups and can also accept backups too), but the search continues!

Otherwise the 'Server' docker web app is great even though the UI could do with a modern look, but Kudos where they're due

**Comment by Danstiv** [comment]
> but the search continues!
restic + backrest?

**Comment by janreinhardt** [comment]
tried Duplicati, tried Kopia, tried Backrest, did not try Borg, think Zerobyte is best

**Comment by lexxxel** [comment]
> tried Duplicati, tried Kopia, tried Backrest, did not try Borg, think Zerobyte is best

Such posts do not help and are unrelated to the issue discussed here. To add context to the state of zerobyte, I quote from the GitHub Readme.md:

⚠️ Warning

Zerobyte is still in version 0.x.x and is subject to major changes from version to version. I am developing the core features and collecting feedback. Please open issues for bugs or feature requests.

---

please stay on topic!

**Comment by janreinhardt** [comment]
This is very much related, since this topics discussion brought us to this point. And you are taking part in it with the „version“ context: Zerobyte is v0.36 wich is isnt different from Kopia v0.23

**Comment by lexxxel** [comment]
> ... since this topics discussion brought us to this point.

Nah, you did. You added other projects to the conversation. Other people only said that they switched over from duplicati and others. You started to compare kopia and recommend other, unstable, software in the kopia bug tracker.

> And you are taking part in it with the „version“ context: Zerobyte is v0.36 wich is isnt different from Kopia v0.23

Both projects do use different versioning schema. Meaning to compare versions by number alone does not say anything about the state of the projects. The amount of commits, the age of the project, the note about 'stability' on the zerobyte repo and the absence of such note on kopia's website should give you a hint. 

Again, it's nice that other projects do stuff that people like, this here is a issue tracker for kopia.

**Comment by Suxsem** [comment]
+1 from me too. Biggest thing missing so far

**Comment by strech345** [comment]
is there any plan to add it?
It sounds like a standard function of backup tools to me and a wish of many users. 

I come from duplicati and was searching for new backup tools to have more stability and higher performance, but also as same comfort. I dont not want to remember where my repos are lying (UI should help here). 
I need to mount my plats first.  i add a description per backup into duplicati to give me the correct mount command. Thats helps me a lot. Also i can clone configs and update it, if something need to be changed.

I understand that this tool did not have this heavy db like duplicati need it, this is cool. But like i understood this is also not needed for this feature because the configs are lying in the kopia app folder as files. So why not showing it?

I would like to have at least a overview over all repos, if also possible on next level a list of backups inside that repos.

For me right now a missing generic feature which let me stay at duplicati.

**Comment by randomragenet** [comment]
Another vote for this. I'm planning on having multiple backup targets that I'd like to automate and I don't want to have to switch back and forth. It'd make sense if I could stay "connected" to both repos and just switch back and forth between them, possibly with a dropdown menu from the UI.

**Comment by CupCakeR** [comment]
For anyone tracking this to set up a 3-2-1 backup strategy (local + cloud): you can use the built-in [Repository Synchronization (`sync-to`)](https://kopia.io/docs/advanced/synchronization/) as a workaround.

**Comment by drewlsvern** [comment]
If there is an appetite from the maintainers to accept a PR for this - I may have something. It is a fairly large change from what I (with Claude's assistance) :)  have done so far.

<img width="3303" height="580" alt="Image" src="https://github.com/user-attachments/assets/8fbaee5b-d3eb-4cfa-ad6f-22e0bfc897aa" />

**Comment by janreinhardt** [comment]
Im already lovin it!

**Comment by replicaro** [comment]
Not to hijack this request, but if anyone really wants multiple repository support from the UI - Replicaro (which uses Kopia under-the-hood) supports multiple repositories: https://github.com/replicaro/replicaro
